### PR TITLE
Fix/tr 299/security issue with execution

### PR DESCRIPTION
**Pull request by tikhanovichA**
https://oat-sa.atlassian.net/browse/TR-299

How to test:
  1) run a test
  2) using the mitmproxy for example, change serviceCallID for next endpoints:
    a) taoQtiTest/Runner/getItem
    b) taoQtiTest/Runner/flagItem

### CODE REVIEW

File: actions/class.Runner.php [hunk 1]
```diff
@@ -128,11 +130,16 @@ protected function getServiceContext()
     {
         if (!$this->serviceContext) {
             $testExecution = $this->getSessionId();
+            /** @var DeliveryExecution $execution */
             $execution = $this->getServiceLocator()->get(DeliveryExecutionService::SERVICE_ID)->getDeliveryExecution($testExecution);
+            $userIdentifier = common_session_SessionManager::getSession()->getUser()->getIdentifier();
```
Comment (mike-ionut-mihai-sandu-tao) [inline actions/class.Runner.php:135]: null point exception might occur, do you think we should check if this exists ?

File: actions/class.Runner.php [hunk 2]
```diff
@@ -26,7 +26,9 @@
 
 use oat\libCat\exception\CatEngineConnectivityException;
 use oat\oatbox\event\EventManager;
+use oat\tao\model\resources\ResourceAccessDeniedException;
```
Comment (mike-ionut-mihai-sandu-tao) [inline actions/class.Runner.php:29]: unused import

File: actions/class.Runner.php (hunk 1 above)
Comment (mike-ionut-mihai-sandu-tao) [inline actions/class.Runner.php:135]: do you think we can have a better way of getting the user Identifier ?

**Review by mike-ionut-mihai-sandu-tao — CHANGES_REQUESTED** [review]

**Comment by mike-ionut-mihai-sandu-tao** [comment]
PS : i think we need to back-port this to TAO Community Edition #134

File: actions/class.Runner.php (hunk 1 above)
Comment (SashaTAO) [inline actions/class.Runner.php]: We can use SessionService here:
`$userIdentifier = $this->getServiceLocator()->get(SessionService::SERVICE_ID)->getCurrentUser()->getIdentifier();`

Comment (gabrielfs7) [inline actions/class.Runner.php:135]: Using this static call is really the only way? Could not we get from serviceLocator instead / also is there any test for this?

File: actions/class.Runner.php [hunk 3]
```diff
@@ -128,11 +130,16 @@ protected function getServiceContext()
     {
         if (!$this->serviceContext) {
             $testExecution = $this->getSessionId();
+            /** @var DeliveryExecution $execution */
             $execution = $this->getServiceLocator()->get(DeliveryExecutionService::SERVICE_ID)->getDeliveryExecution($testExecution);
+            $userIdentifier = common_session_SessionManager::getSession()->getUser()->getIdentifier();
+            if ($execution->getUserIdentifier() !== $userIdentifier) {
```
Comment (SashaTAO) [inline actions/class.Runner.php]: Maybe better to extract into small method, something like `checkAuthorization`, `isAuthorize` or something like that.

**Review by SashaTAO — CHANGES_REQUESTED** [review]

File: actions/class.Runner.php [hunk 4]
```diff
@@ -259,12 +269,11 @@ protected function getErrorCode($e = null)
...
-        $serviceContext = $this->getRunnerService()->initServiceContext($this->getServiceContext());
-
         $this->checkSecurityToken();
 
         try {
+            /** @var QtiRunnerServiceContext $serviceContext */
+            $serviceContext = $this->getRunnerService()->initServiceContext($this->getServiceContext());
             $this->returnJson($this->getInitResponse($serviceContext));
         } catch (Exception $e) {
             $this->returnJson(
```
Comment (zagovorichev) [inline actions/class.Runner.php:279]: As it is general Exception - could you at least store Log::e($e->getMessage())?

**Review by zagovorichev — CHANGES_REQUESTED** [review]

File: actions/class.Runner.php (hunk 4 above)
Comment (tikhanovichA) [reply actions/class.Runner.php:279]: agree

Comment (tikhanovichA) [reply actions/class.Runner.php:280]: I also added logs for all exceptions where it was missed

**Review by zagovorichev — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: actions/class.Runner.php [hunk 5]
```diff
@@ -292,6 +302,7 @@ public function getTestData()
                 'success' => true,
             ];
         } catch (common_Exception $e) {
+            common_Logger::e($e->getMessage());
```
Comment (boajer) [inline actions/class.Runner.php:305]: Please avoid static logger calls here and bellow

File: views/package-lock.json [hunk 6]
```diff
@@ -7,7 +7,7 @@
     "@oat-sa/tao-test-runner": {
       "version": "0.5.0",
       "resolved": "https://registry.npmjs.org/@oat-sa/tao-test-runner/-/tao-test-runner-0.5.0.tgz",
-      "integrity": "sha512-TBUeLGRdqYChIDteImjTXcMYKOxTvho/hAuRMyFyBlTlpwNsGt1PFgTKevMP8g7MadHTcKXmPHK0unEanNHD0Q=="
+      "integrity": "sha1-N/j4WxVgkYN9sjZDov4yOqpABhg="
```
Comment (boajer) [inline views/package-lock.json:10]: Why you've pushed this?

**Review by boajer — APPROVED** [review]

**Pull request merged**